kfs-core/src/main/java/edu/arizona/kfs/rq/batch/helper/impl/DocumentRequeueReportWriterTextImpl.java

```diff
@@ -0,0 +1,188 @@
+package edu.arizona.kfs.rq.batch.helper.impl;
+
+import edu.arizona.kfs.rq.batch.helper.DocumentRequeueReportWriter;
+import org.apache.commons.lang.StringUtils;
+import org.kuali.rice.core.api.datetime.DateTimeService;
+import org.kuali.rice.krad.util.ObjectUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.Date;
+
+/**
+ * UAF-475: MOD-WKFLW-03 Requeue Stuck Documents Job
+ *
+ * @author Josh Shaloo <shaloo@email.arizona.edu>
+ */
+public class DocumentRequeueReportWriterTextImpl implements DocumentRequeueReportWriter {
+
+    protected String reportDirectoryName;
+    protected String reportDate;
+    protected String outputFileName;
+
+    protected final String REPORT_FILENAME = "RequeuedDocuments";
+    protected final String REPORT_FILE_SUFFIX = ".txt";
+
+    protected final String REPORT_STATUS_HEADER_TEXT = "Status";
+    protected final String REPORT_DOCNUM_HEADER_TEXT = "Doc Num";
+    protected final String REPORT_DOCTYPE_HEADER_TEXT = "Doc Type";
+    protected final String REPORT_MODIFIED_HEADER_TEXT = "Last Modified";
+    protected final String REPORT_DOCUMENT_TITTLE_HEADER_TEXT = "Document Title";
+    protected final String REPORT_BLANK_SPACE_TEXT = " ";
+
+    protected final Integer REPORT_STATUS_COLUMN_LENGTH = 9;
+    protected final Integer REPORT_DOCNUM_COLUMN_LENGTH = 9;
+    protected final Integer REPORT_DOCTYPE_COLUMN_LENGTH = 40;
+    protected final Integer REPORT_LAST_MODIFIED_COLUMN_LENGTH = 30;
+    protected final Integer REPORT_DOCUMENT_TITLE_COLUMN_LENGTH = 60;
+    protected final Integer REPORT_BUFFER_LENGTH = 200;
+
+    private StringBuffer reportLineBuffer;
+    private File reportFile;
+    private PrintStream outputStream;
+    private DateTimeService dateTimeService;
+
+    public DocumentRequeueReportWriterTextImpl ( String reportDirectoryName, DateTimeService dateTimeService ) {
+        setReportDirectoryName(reportDirectoryName);
+        setDateTimeService(dateTimeService);
+    }
+
+    public void initializeReport() {
+
+        File reportFile = getReportFile();
+
+        try {
+            reportFile.createNewFile();
+            this.outputStream = new PrintStream(reportFile);
+        } catch (IOException ex) {
+            finalizeReport();
+            throw new RuntimeException("Failed to open report for writing.", ex);
+        }
+
+        printReportHeader();
+    }
+
+
+
+    public void reportRequeueStatus( String requeueStatus, String documentNumber, String documentType, String lastModified, String lastAction ) {
+        printRow(
+                requeueStatus,
+                documentNumber,
+                documentType,
+                lastModified,
+                lastAction
+        );
+    }
+
+    public void finalizeReport() {
+        PrintStream outputStream = getOutputStream();
+
+        if (ObjectUtils.isNotNull(outputStream)) {
```

Comment:
A nice way to get around bothersome stream cleanup, is apache's "IOUtils.closeQuietly(stream)". This lib is already included in the project.